nova/scheduler/filters/affinity_filter.py

```diff
@@ -32,15 +32,22 @@ class DifferentHostFilter(filters.BaseHostFilter):
     RUN_ON_REBUILD = False
 
     def host_passes(self, host_state, spec_obj):
-        affinity_uuids = spec_obj.get_scheduler_hint('different_host')
+        affinity_uuids = self.host_info_requiring_instance_ids(spec_obj)
         if affinity_uuids:
             overlap = utils.instance_uuids_overlap(host_state, affinity_uuids)
             return not overlap
         # With no different_host key
         return True
 
     def host_info_requiring_instance_ids(self, spec_obj):
-        return set(spec_obj.get_scheduler_hint('different_host'))
+        different_host = spec_obj.get_scheduler_hint('different_host')
+        if not different_host:
+            return different_host
```

Comment:
Do you want to return a `set` always? If you do, then I'd do it explicitly, or else risk returning `None` or even an empty string:
```suggestion
            return set()
```